spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -138,3 +138,48 @@ When a stream is comprised of multiple modules with the same name, they must be
 
   stream create --definition "http | firstLabel: transform --expression=payload.toUpperCase() | secondLabel: transform --expression=payload+'!' | log" --name myStreamWithLabels --deploy
 
+[[tap-dsl]]
+== Tap DSL
+
+Taps can be created at various producer endpoints in a stream. For a stream like this:
+
+  stream create --definition "http | step1: transform --expression=payload.toUpperCase() | step2: transform --expression=payload+'!' | log" --name mainstream --deploy
+
+taps can be created at the output of `http`, `step1` and `step2`.
+
+Tap DSL requires to specify the `source destination name` for the tap stream. The syntax for source destination name is:
+
+`<stream-name>.<label/app name>`
+
+To create a tap at the output of `http` in the stream above, the source destination name is `mainstream.http`
+To create a tap at the output of first transform app in the stream above, the source destination name is `mainstream.step1`
+
+The tap stream DSL looks like this:
+
+  stream create --definition ":mainstream.http > counter" --name tap_at_http --deploy
+
+  stream create --definition ":mainstream.step1 > jdbc" --name tap_at_step1_transformer --deploy
+
+Note the colon (:) prefix before the destination names. The colon lets the parser to parse this as the destination name instead of app name.
+
+[[explicit-destination-names]]
+== Connecting to explicit destination names at the broker
+
+One can connect to specific source destination name located in the broker (Rabbit, Kafka) and connect it to the downstream.
+In this case, the DSL looks like this:
+
+  stream create --definition ":mySource > log" --name ingest_from_broker --deploy
+
+In the above stream, the destination name `mySource` is the `source` destination name and stream receives messages from the source destination `mySource` at the broker and connect it to the `log` app.
+
+The destination can act as `sink` as well.
+
+  stream create --definition "http > :mySink" --name ingest_to_broker --deploy
+
+In the above stream, the destination name `mySink` is the `sink` destination name and the stream sends the messages from `http` app to the destination `mySink` at the broker.
+
+It is also possible to connect `source` and `sink` destinations at the broker.
+
+  stream create --definition ":myOtherSource > :myOtherSink" --name source_sink_destinations --deploy
+
+In the above stream, both the source and sink destinations are located in the broker and messages flow from the source destination to the sink destination.
```

Comment:
This sentence seems to imply that the flow happens entirely at the broker level, while there are actually two hops and an implicit bridge app involved. I think we should clarify this